tests/expected/offset-invalid-args/invalid_args.expected

```diff
@@ -0,0 +1 @@
+Cannot offset non-pointer type
```

Comment:
Where does this error message come from? I would have expected to see the message from `validate_raw_ptr`.
Also, can you test that `validate_offset` catches failures too?

Reply:
So, the Rust compiler actually crashes before codegen. This is a panic message from the compiler.

I added the validation and tests just in case they eventually decide to handle this case differently, but today, the validations are redundant.